user-interface/0001-WALL-wallet.md

```diff
@@ -0,0 +1,179 @@
+
+## Taint keys
+When protecting myself from use of keys that may be compromised, I..
+
+- **must** select a key I wish to taint <a name="0001-WALL-057" href="#0001-WALL-057">0001-WALL-057</a>
+- **must** be prompted to enter wallet password to taint key <a name="0001-WALL-058" href="#0001-WALL-058">0001-WALL-058</a>
+- (Dapps that request use of tainted keys **must** not prompt a prompt user to sign transaction) <a name="0001-WALL-059" href="#0001-WALL-059">0001-WALL-059</a>
+- **must** see tainted keys flagged as tainted <a name="0001-WALL-060" href="#0001-WALL-060">0001-WALL-060</a>
+
+... so that tainted keys must not be used
+
+When I have accidentally tainted a key I...
+
+- **must** select a key to un-taint by entering wallet password <a name="0001-WALL-061" href="#0001-WALL-061">0001-WALL-061</a>
+
+...so that I must use the key again
+
+## Isolate keys
+See [docs on key isolation](https://docs.vega.xyz/docs/mainnet/tools/vega-wallet/cli-wallet/latest/guides/isolate-keys)
+When I want to create an extra level of security for a given key, I...
+
+- `TODO` must select a key that I want to isolate
+- `TODO` am prompted for a password before isolation
+- `TODO` am instructed as to where the wallet file has been created
+
+... so I must store some keys in an extra secure way
+
+## Manually sign a message
+When wishing to use my wallet to sign arbitrary messages, I...
+
+- **must** enter content to be signed with key  <a name="0001-WALL-062" href="#0001-WALL-062">0001-WALL-062</a>
+- **must** see an option to base64 encode content before signing <a name="0001-WALL-063" href="#0001-WALL-063">0001-WALL-063</a>
+- **must** see option to broadcast that to a selected network <a name="0001-WALL-064" href="#0001-WALL-064">0001-WALL-064</a>
+- **must** be able to submit/sign the content and am given a hash of the signed content as well as the message (now encoded) <a name="0001-WALL-065" href="#0001-WALL-065">0001-WALL-065</a>
+  - **must** be able to [track progress](0003-WTXN-submit_vega_transaction.md#track-transaction-on-network) of broadcast transaction <a name="0001-WALL-071" href="#0001-WALL-071">0001-WALL-071</a>
+
+.. so I can control of the message being signed, and can use the message elsewhere (for example to prove I own a wallet)
+
+## Wallet management
+When seeking to reduce risk of compromise I...
+
+- **must** be able to create multiple wallets <a name="0001-WALL-066" href="#0001-WALL-066">0001-WALL-066</a>
+- **must** be able to switch between wallets <a name="0001-WALL-067" href="#0001-WALL-067">0001-WALL-067</a>
+- **must** be able to delete a wallet <a name="0001-WALL-068" href="#0001-WALL-068">0001-WALL-068</a>
+- **must** be able to change wallet name <a name="0001-WALL-069" href="#0001-WALL-069">0001-WALL-069</a>
+- **must** be able to change wallet passphrase <a name="0001-WALL-070" href="#0001-WALL-070">0001-WALL-070</a>
+- `TODO` must get the recovery phrase for a wallet (at any time not just creation)
```

Comment:
I want this but have been told many times  this is a security risk - but it would be nice to have a way to make people interact with their recovery phrase or double-agree that they've saved it somewhere safe/secure.